### PR TITLE
Overview team in project — crew integration

### LINKED ISSUE #2215: Overview team in project — crew integration
What:
- team title
- people count
- link to team page

How:
- field with team id in project settings

### PR DESCRIPTION
**Pull request by asabotovich**
## PR includes

- [x] Feature

## Related issues

Resolve #2215 

## QA Instructions, Screenshots, Recordings
  

https://github.com/taskany-inc/issues/assets/9020793/11251450-ad83-439c-8298-040392020b52

### CODE REVIEW

**Comment by asabotovich** [comment]
Screenshot 2024-02-16 at 15 54 33

File: src/components/ProjectTeamPage/ProjectTeamPage.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+    if (!project) return null;
+
+    const pageTitle = tr
+        .raw('title', {
+            project: project.title,
+        })
+        .join('');
+
+    console.log('QQQ TEAMS', teams);
```
Comment (tsumo) [inline src/components/ProjectTeamPage/ProjectTeamPage.tsx:58]: :)

**Comment by asabotovich** [comment]
https://github.com/taskany-inc/issues/assets/9020793/6c020c2a-f48b-46be-82d7-20be31c6c1c3

File: src/components/TeamComboBox.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            },
+            [updateProject, project],
+        );
+
+        const resetSearch = useCallback(() => {
+            setSearch('');
+        }, []);
+
+        return (
+            <ComboBox
```
Comment (LamaEats) [inline src/components/TeamComboBox.tsx:58]: Why legacy component instead of Dropdown and AutoComplete from `harmony`?

File: prisma/schema.prisma [hunk 3]
```diff
@@ -163,6 +163,7 @@ model Project {
   personal     Boolean?   @default(false)
   archived     Boolean?   @default(false)
   sharedGoals  Goal[]     @relation("partnershipProjects")
+  teams        String[]
```
Comment (LamaEats) [inline prisma/schema.prisma:166]: maybe call as `externalTeams`? Because get it from other service

File: src/types/crew.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+export interface Team {
+    name: string;
+    description?: string;
+    id: string;
+    memberships: Membership[];
+}
+
+interface Membership {
```
Comment (LamaEats) [inline src/types/crew.ts:8]: define before usage

File: src/components/ProjectTeamPage/ProjectTeamPage.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
+import { nullable, TreeView, TreeViewElement, TreeViewNode } from '@taskany/bricks';
```
Comment (awinogradov) [inline src/components/ProjectTeamPage/ProjectTeamPage.tsx:1]: harmony/TreeView

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/components/TeamComboBox.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
+import React, { useCallback, useState, ChangeEvent } from 'react';
+import { ComboBox } from '@taskany/bricks';
```
[the remark this answers is not in the document]
Comment (asabotovich) [reply src/components/TeamComboBox.tsx]: fixed

File: src/components/TeamListItem.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import { FC, useCallback } from 'react';
+import { Button, Text } from '@taskany/bricks/harmony';
+import { IconBinOutline } from '@taskany/icons';
+import { TableCell } from '@taskany/bricks';
```
[the remark this answers is not in the document]
Comment (asabotovich) [reply src/components/TeamListItem.tsx:4]: will be fixed here https://github.com/taskany-inc/issues/issues/2266

File: src/types/crew.ts (hunk 4 above)
Comment (asabotovich) [reply src/types/crew.ts]: fixed

**Comment by asabotovich** [comment]
telegram-cloud-photo-size-2-5379648148756091773-y

**Review by tsumo — APPROVED** [review]

**Pull request merged**